### PR TITLE
Clean-up waffle-221 as per review comments.

### PR DESCRIPTION
**Pull request by joesoftware**

### CODE REVIEW

**Comment by alexose** [comment]
Is waffle-221 the same thing as https://github.com/department-of-veterans-affairs/healthcare-application-team/issues/221?

**Comment by joesoftware** [comment]
Alex, this is the clean-up from Albert's comments on waffle-221... which I made in a newer branch.

File: src/server/enrollment-system.js [hunk 1]
```diff
@@ -63,22 +63,163 @@ function veteranToPersonInfo(veteran) {
...
+ * Warning: If 'number' is wider than 'width' characters, only the right-most 'width'
+ * characters are returned.
+ *
+ * TODO: should this explicitly check if 'number' and 'width' are actually numbers?
+ *
+ * @param {number} number The number you want left-padded with '0' characters.
+ * @param {number} width The width of the resulting string.
+ * @returns {String}
+ *
+ * @example
```
Comment (awong-dev) [inline src/server/enrollment-system.js]: @example usually goes before the @param and @return blobs.  The param + returns are often the last thing in the comment block IIRC.

File: src/server/enrollment-system.js [hunk 2]
```diff
@@ -63,22 +63,163 @@ function veteranToPersonInfo(veteran) {
...
+ * // returns false
+ * isLeapYear(2011)
+ *
+ * @example
+ * // returns [ true, false, false, false, true, true, true ]
+ * [1600,1700,1899,1900,2000,2016,2012].map(isLeapYear)
+ */
+function isLeapYear(year) {
+  return ((year % 4 === 0) && (year % 100 !== 0) || (year % 400 === 0));
```
Comment (awong-dev) [inline src/server/enrollment-system.js]: Can we document where we got this logic?

File: src/server/enrollment-system.js [hunk 3]
```diff
@@ -63,22 +63,163 @@ function veteranToPersonInfo(veteran) {
...
+ *
+ * @example
+ * // returns false
+ * isLeapYear(2011)
+ *
+ * @example
+ * // returns [ true, false, false, false, true, true, true ]
+ * [1600,1700,1899,1900,2000,2016,2012].map(isLeapYear)
+ */
+function isLeapYear(year) {
```
Comment (awong-dev) [inline src/server/enrollment-system.js]: Can you add a TODO in here and file an issue against me to double check the date logic?  This stuff always scares me.

**Comment by awong-dev** [comment]
LGTM w nits.

Mostly some comment ordering changes, but stamping LGTM.  Thanks for fixing!

Feel free to merge after you resolve and it passes travis.

File: src/server/enrollment-system.js (hunk 1 above)
Comment (awong-dev) [inline src/server/enrollment-system.js]: Also, I think the examples here might be overkill. The APIs are pretty self-explanitory...

File: src/server/enrollment-system.js [hunk 4]
```diff
@@ -37,19 +37,169 @@ const formTemplate = {
...
   return (new Array(padding + 1).join('0') + number).slice(-padding);
 }
 
-// TODO(awong): Move to validations and add unittests.
-function formDateToESDate(dateObject) {
-  if (dateObject.month >= 1 && dateObject.month <= 12 &&
-    dateObject.day >= 1 && dateObject.month <= 31 && // TODO: how robust does this need to be? Form validates as well.
-    dateObject.year > 1) {
-    return `${zeroPadNumber(dateObject.month, 2)}/${zeroPadNumber(dateObject.day, 2)}/${zeroPadNumber(dateObject.year, 4)}`;
+/**
```
Comment (alexose) [inline src/server/enrollment-system.js]: We should be able to use moment.js for most of the following:

File: src/server/enrollment-system.js [hunk 5]
```diff
@@ -37,19 +37,169 @@ const formTemplate = {
...
+ *   [ 2017, false ] ]
+ *
+ * [1700,1800,1900,2000,2011,2012,2013,2014,2015,2016,2017].map((a)=>{return [a,isLeapYear(a)]})
+ *
+ * TODO: Albert, please double check the date logic.
+ *
+ * @param {number} year The year you want to test.
+ * @returns {boolean}
+ */
+function isLeapYear(year) {
```
Comment (alexose) [inline src/server/enrollment-system.js]: https://github.com/moment/momentjs.com/blob/master/docs/moment/05-query/09-is-leap-year.md

File: src/server/enrollment-system.js [hunk 6]
```diff
@@ -37,19 +37,169 @@ const formTemplate = {
...
+ * formDateToESDate({month: 1, day: 30, year: 1980})
+ *
+ * @example
+ * // returns undefined
+ * formDateToESDate(100)
+ *
+ * @param {Object} dateObject in the format of {month: 1, day: 30, year: 1980}
+ * @returns {String}
+ */
+function formDateToESDate(dateObject) {
```
Comment (alexose) [inline src/server/enrollment-system.js]: ```moment(dateObject).format('MM/DD/YYYY')```

**Comment by alexose** [comment]
Code looks good to me! 👍

File: src/server/enrollment-system.js (hunk 6 above)
Comment (awong-dev) [inline src/server/enrollment-system.js]: That actually doesn't work..... or it didn't. robbie had that previously.

**Pull request merged**